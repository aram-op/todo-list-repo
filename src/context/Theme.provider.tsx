'use client';

import React, {useContext, useState} from 'react';
import {Theme, ThemeContext} from '@/context/Theme.context';

export type ThemeProviderProps = {
    children: React.ReactNode;
};

function ThemeContextProvider({children}: ThemeProviderProps) {
    const [theme, setTheme] = useState(Theme.LIGHT);

    const toggleTheme = () =>
        setTheme((oldTheme) =>
            oldTheme === Theme.LIGHT ? Theme.DARK : Theme.LIGHT
        );

    return (
        <ThemeContext.Provider value={{theme, toggleTheme}}>
            <div className={theme}>{children}</div>
        </ThemeContext.Provider>
    );
};

const useThemeContext = () => {
    const context = useContext(ThemeContext);

    if (context === undefined) {
        throw new Error(
            'useThemeContext must be used within a ThemeContextProvider'
        );
    }

    return context;
};

export {ThemeContextProvider, useThemeContext};