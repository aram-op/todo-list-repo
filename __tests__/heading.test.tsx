import {fireEvent, render, screen} from '@testing-library/react';
import {Theme, ThemeContext} from '@/context/Theme.context';
import Heading from '@/app/ui/heading/heading';

describe('Heading component', () => {
    it('should renders the heading', () => {
        render(
            <ThemeContext.Provider value={{ toggleTheme: jest.fn(), theme: Theme.LIGHT }}>
                <Heading />
            </ThemeContext.Provider>
        );

        expect(screen.getByText('Todoist')).toBeInTheDocument();

        expect(screen.getByAltText('todoist logo')).toBeInTheDocument();
        expect(screen.getByAltText('sun icon')).toBeInTheDocument();
        expect(screen.getByAltText('moon icon')).toBeInTheDocument();
    });

    it('should call .toggleTheme if the button was clicked', () => {
        const mockToggleTheme = jest.fn();

        render(
            <ThemeContext.Provider value={{ toggleTheme: mockToggleTheme, theme: Theme.LIGHT }}>
                <Heading />
            </ThemeContext.Provider>
        );

        const button = screen.getByRole('button');
        fireEvent.click(button);

        expect(mockToggleTheme).toHaveBeenCalled();
    });

    it('should match snapshot', () => {
        const {asFragment} = render(
            <ThemeContext.Provider value={{ toggleTheme: jest.fn(), theme: Theme.LIGHT }}>
                <Heading />
            </ThemeContext.Provider>
        );

        expect(asFragment).toMatchSnapshot();
    });
});
