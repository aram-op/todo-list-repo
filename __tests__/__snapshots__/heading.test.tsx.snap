// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component should match snapshot 1`] = `[Function]`;
